stack/scripts/create.sh

```diff
@@ -20,40 +18,69 @@ if [[ $BASH_VERSINFO -lt 4 ]]; then
 fi
 
 function main() {
-  local flags
+  local flags stack_dir_name build_dir_name
+  stack_dir_name=""
+  build_dir_name=""
 
   while [[ "${#}" != 0 ]]; do
     case "${1}" in
-      --help|-h)
-        shift 1
-        usage
-        exit 0
-        ;;
-
-      --secret)
-        flags+=("--secret" "${2}")
-        shift 2
-        ;;
-
-      --label)
-        flags+=("--label" "${2}")
-        shift 2
-        ;;
-
-      "")
-        # skip if the argument is empty
-        shift 1
-        ;;
-
-      *)
-        util::print::error "unknown argument \"${1}\""
+    --help | -h)
+      shift 1
+      usage
+      exit 0
+      ;;
+
+    --secret)
+      flags+=("--secret" "${2}")
+      shift 2
+      ;;
+
+    --label)
+      flags+=("--label" "${2}")
+      shift 2
+      ;;
+
+    --stack-dir)
+      stack_dir_name="${2}"
+      shift 2
+      ;;
+
+    --build-dir)
+      build_dir_name="${2}"
+      shift 2
+      ;;
+
+    "")
+      # skip if the argument is empty
+      shift 1
+      ;;
+
+    *)
+      util::print::error "unknown argument \"${1}\""
+      ;;
     esac
   done
 
-  mkdir -p "${BUILD_DIR}"
-
   tools::install
-  stack::create "${flags[@]}"
+
+  #if stack or build argument is provided but not both, then throw an error
+  if [[ -n "${stack_dir_name}" && ! -n "${build_dir_name}" ]] || [[ ! -n "${stack_dir_name}" && -n "${build_dir_name}" ]]; then
+    util::print::error "Both stack-dir and build-dir must be provided"
+  elif [[ -n "${stack_dir_name}" && -n "${build_dir_name}" ]]; then
+    stack::create "${ROOT_DIR}/${stack_dir_name}" "${ROOT_DIR}/${build_dir_name}" "${flags[@]}"
+  elif [ -f "${ROOT_DIR}/stacks.json" ]; then
+    stack_names=($(jq -r '.[] ' "${ROOT_DIR}/stacks.json"))
```

Comment:
Discussed with Costas and we are going tweak the format of the stacks.json to make it a bit more general.

Also since stacks are going away at some point, but we will still need to create build and run images,  maybe we should call this images.json to start.